ipc/common/messages.go

```diff
@@ -0,0 +1,217 @@
+	messageKindNameByID[BlockchainCurrentNonceRequest] = "BlockchainCurrentNonceRequest"
+	messageKindNameByID[BlockchainCurrentNonceResponse] = "BlockchainCurrentNonceResponse"
+	messageKindNameByID[BlockchainCurrentRoundRequest] = "BlockchainCurrentRoundRequest"
+	messageKindNameByID[BlockchainCurrentRoundResponse] = "BlockchainCurrentRoundResponse"
+	messageKindNameByID[BlockchainCurrentTimeStampRequest] = "BlockchainCurrentTimeStampRequest"
+	messageKindNameByID[BlockchainCurrentTimeStampResponse] = "BlockchainCurrentTimeStampResponse"
+	messageKindNameByID[BlockchainCurrentRandomSeedRequest] = "BlockchainCurrentRandomSeedRequest"
+	messageKindNameByID[BlockchainCurrentRandomSeedResponse] = "BlockchainCurrentRandomSeedResponse"
+	messageKindNameByID[BlockchainCurrentEpochRequest] = "BlockchainCurrentEpochRequest"
+	messageKindNameByID[BlockchainCurrentEpochResponse] = "BlockchainCurrentEpochResponse"
+	messageKindNameByID[DiagnoseWaitRequest] = "DiagnoseWaitRequest"
+	messageKindNameByID[DiagnoseWaitResponse] = "DiagnoseWaitResponse"
+	messageKindNameByID[LastKind] = "LastKind"
+}
+
+// MessageHandler is a message abstraction
+type MessageHandler interface {
+	GetNonce() uint32
+	SetNonce(nonce uint32)
+	GetKind() MessageKind
+	SetKind(kind MessageKind)
+	GetError() error
+	SetError(err error)
+}
+
+// Message is the implementation of the abstraction
+type Message struct {
+	DialogueNonce uint32
+	Kind          MessageKind
+	ErrorMessage  string
+}
+
+// GetNonce gets the dialogue nonce
+func (message *Message) GetNonce() uint32 {
+	return message.DialogueNonce
+}
+
+// SetNonce sets the dialogue nonce
+func (message *Message) SetNonce(nonce uint32) {
+	message.DialogueNonce = nonce
+}
+
+// GetKind gets the message kind
+func (message *Message) GetKind() MessageKind {
+	return message.Kind
+}
+
+// SetKind sets the message kind
+func (message *Message) SetKind(kind MessageKind) {
+	message.Kind = kind
+}
+
+// GetError gets the error within the message
+func (message *Message) GetError() error {
+	if message.ErrorMessage == "" {
+		return nil
+	}
+
+	return fmt.Errorf(message.ErrorMessage)
+}
+
+// SetError sets the error within the message
+func (message *Message) SetError(err error) {
+	if err != nil {
+		message.ErrorMessage = err.Error()
+	}
+}
+
+func (message *Message) String() string {
+	kindName, _ := messageKindNameByID[message.Kind]
+	return fmt.Sprintf("[kind=%s nonce=%d err=%s]", kindName, message.DialogueNonce, message.ErrorMessage)
+}
+
+// MessageStop is a message sent by Node to stop Arwen
+type MessageStop struct {
+	Message
+}
+
+// NewMessageStop creates a new message
+func NewMessageStop() *MessageStop {
+	message := &MessageStop{}
+	message.Kind = Stop
+	return message
+}
+
+// MessageReplier is a callback signature
+type MessageReplier func(MessageHandler) MessageHandler
+
+func noopReplier(message MessageHandler) MessageHandler {
+	panic("NO-OP replier called")
```

Comment:
do we really want panic ?

Reply:
Fixed, now logging an error. Refactored so that `CreateReplySlots` receives a noop replier.